src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -366,33 +395,40 @@ ColumnsDescription StorageObjectStorage::resolveSchemaFromData(
     const ObjectStoragePtr & object_storage,
     const ConfigurationPtr & configuration,
     const std::optional<FormatSettings> & format_settings,
+    std::string & sample_path,
     const ContextPtr & context)
 {
     ObjectInfos read_keys;
     auto iterator = createReadBufferIterator(object_storage, configuration, format_settings, read_keys, context);
+    sample_path = iterator->getLastFilePath();
     return readSchemaFromFormat(configuration->format, format_settings, *iterator, context);
```

Comment:
Similar here we need to call `sample_path = iterator->getLastFilePath();` after `readSchemaFromFormat`